From 28f07715ab4d670ce81e12776bbece043305bd83 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 11 Apr 2020 15:08:39 -0700
Subject: [PATCH] server: Fix build when printf is a macro

clang complains on x86 when building

main.c:116:2: error: embedding a #include directive within macro arguments is not supported
 ^

convert nesting include into a string assignment, to same effect but
making it compatible with clang as well

Upstream-Status: Submitted [https://github.com/libguestfs/nbdkit/pull/3]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 server/main.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/server/main.c b/server/main.c
index 748122f..c0ac874 100644
--- a/server/main.c
+++ b/server/main.c
@@ -112,9 +112,10 @@ static void
 usage (void)
 {
   /* --{short,long}-options remain undocumented */
-  printf (
+  char const *opt_list =
 #include "synopsis.c"
-  );
+	  ;
+  printf ("%s\n", opt_list);
   printf ("\n"
           "Please read the nbdkit(1) manual page for full usage.\n");
 }
-- 
2.26.0

